### PR TITLE
Add chrome on android to CI tests

### PR DESCRIPTION
**Pull request by tafsiri**
This adds a chrome on android to our test matrix.

It includes fixes to get `isNaN` and this NaN propagation working. As well as fix to concat that was occurring on Pixel 3 (this was mostly likely due to a compiler/driver issue not the original code). 

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2015)

### CODE REVIEW

**Comment by tafsiri** [comment]
Updating ios version didn't work. os_version strings advertised by browserstack rest API errored out at trying to start the device.

File: tfjs-core/src/backends/webgl/concat_packed_gpu.ts [hunk 1]
```diff
@@ -72,18 +78,47 @@ export class ConcatPackedProgram implements GPGPUProgram {
...
           result.b = getValue(${coords});
         }
         setOutput(result);
       }
     `;
   }
 }
+
+/**
+ * Return an expression for coordiantes into a vector where a given channel
```
Comment (annxingyuan) [inline tfjs-core/src/backends/webgl/concat_packed_gpu.ts]: nit: coordinates

File: tfjs-core/src/backends/webgl/glsl_version.ts [hunk 2]
```diff
@@ -50,10 +50,22 @@ export function getGlslDifferences(): GLSL {
...
     output = 'outputColor';
     defineOutput = 'out vec4 outputColor;';
+
+    // Use custom isnan definition to work across differences between
+    // implementations on various platforms. While this should happen in ANGLE
+    // we still see differences between android and windows (on chrome) when
+    // using isnan directly.
     defineSpecialNaN = `
       bool isnan_custom(float val) {
-        return (val > 0. || val < 0. || val == 0.) ? false : true;
```
Comment (annxingyuan) [inline tfjs-core/src/backends/webgl/glsl_version.ts]: that's crazy...

File: tfjs-core/src/backends/webgl/concat_packed_gpu.ts (hunk 1 above)
Comment (tafsiri) [reply tfjs-core/src/backends/webgl/concat_packed_gpu.ts]: Done

File: tfjs-core/src/backends/webgl/glsl_version.ts (hunk 2 above)
Comment (tafsiri) [reply tfjs-core/src/backends/webgl/glsl_version.ts]: close in form but slightly different to what we had before!

**Comment by nsthorat** [comment]
This is causing a bunch of timeouts now can you look into this? 

https://pantheon.corp.google.com/cloud-build/builds/96d382d1-9b2a-4395-b883-1ca4727247c8;step=2?project=learnjs-174218

**Pull request merged**